pkg/controllers/certificaterequest_controller.go

```diff
@@ -145,22 +150,26 @@ func (r *CertificateRequestReconciler) Reconcile(ctx context.Context, req ctrl.R
 	iss, err := util.GetIssuer(ctx, r.Client, issuerName)
 	if err != nil {
 		log.Error(err, "failed to retrieve Issuer resource")
-		_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonFailed, "issuer could not be found")
+		_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonPending, "issuer could not be found")
 		return ctrl.Result{}, err
 	}
 
 	if !isReady(iss) {
 		err := fmt.Errorf("issuer %s is not ready", iss.GetName())
-		_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonFailed, "issuer is not ready")
+		_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonPending, "issuer is not ready")
 		return ctrl.Result{}, err
 	}
 
-	provisioner, ok := aws.GetProvisioner(issuerName)
-	if !ok {
-		err := fmt.Errorf("provisioner for %s not found", issuerName)
-		log.Error(err, "failed to retrieve provisioner")
-		_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonFailed, "failed to retrieve provisioner")
-		return ctrl.Result{}, err
+	provisioner, exists := awspca.GetProvisioner(issuerName)
+	if !exists {
+		config, err := awspca.GetConfig(ctx, r.Client, iss.GetSpec())
+		if err != nil {
+			log.Error(err, "error loading config")
+			_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonFailed, "failed to retrieve aws config")
+			return ctrl.Result{}, err
+		}
+
+		provisioner = NewProvisioner(config, iss.GetSpec().Arn, issuerName)
 	}
```

Comment:
Can we just hide all this logic inside of `GetProvisioner`. From the perspective of the `CertificateRequestController` it really only cares that GetProvisoner returns something -- not how.

Reply:
I think the only appropriate part to leave here is the:

```
		if err != nil {
			log.Error(err, "error loading config")
			_ = r.setStatus(ctx, cr, cmmeta.ConditionFalse, cmapi.CertificateRequestReasonFailed, "failed to retrieve aws config")
			return ctrl.Result{}, err
		}
```

which should get run if `GetProvisoner` returns an error.